test/helpers/kubectl.go

```diff
@@ -463,6 +463,41 @@ func (kub *Kubectl) WaitKubeDNS() error {
 	return err
 }
 
+// WaitForKubeDNSEntry waits until the given DNS entry is ready in kube-dns
+// pod. If the container is not ready after timeout it returns an error. The
+// name's format query should be `${name}.${namespace}`. If `svc.cluster.local`
+// is not present it appends to the given name and it checks  the full FQDN
+func (kub *Kubectl) WaitForKubeDNSEntry(name string) error {
+	svcSuffix := "svc.cluster.local"
+	logger := kub.logger.WithField("dnsName", name)
+
+	if !strings.HasSuffix(name, svcSuffix) {
+		name = fmt.Sprintf("%s.%s", name, svcSuffix)
+	}
+	// https://bugs.launchpad.net/ubuntu/+source/bind9/+bug/854705
+	digCMD := "dig +short %s @%s | grep -v -e '^$'"
+
+	podsIPs, err := kub.GetPodsIPs(KubeSystemNamespace, "k8s-app=kube-dns")
```

Comment:
I thought pods were using the kube-dns service cluster IP, and not the kube-dns pod IPs?

Reply:
Yes, but we executed the dig command from the base host. The main reason is because cilium image does not have `host`, `nslookup` or `dig` installed. So we test from base host to the kubedns host to validate that it's ready.

Reply:
Ok.

Reply:
Ok.